Automattic/wp-calypso
# Add basic e2e tests to cover FAB support search instance

#### Changes proposed in this Pull Request

We do not currently have any tests to cover the support search functionality in:

* the "FAB" inline help popover.
* the "Get help" card on My Home.

This PR will **add e2e tests to cover the basic mechanics of the "FAB" inline search _only_**. This provides us with a solid foundation upon which to:

- refactor the "Get help" card to share components with the FAB implementation.
- refactor other aspects of the search components.
- add new functionality (such as being able to search site settings).

**Please note**: this only covers the FAB version as the My Home version is being refactored to use the same components we will wait until they are normalised before adding tests for My Home.

#### Testing instructions

* Configure the Calypso local e2e testing environment [as per the docs](https://github.com/Automattic/wp-calypso/blob/f35d4cc98655259e379fe220be9b9ecee6bac647/test/e2e/README.md).
* Boot Calypso locally - `yarn && yarn start`. Make sure it's running at `http://calypso.localhost:3000/`.
* Run the individual e2e test 
  - `cd` to the `test/e2e` directory.
  - run `./node_modules/.bin/mocha specs/wp-support-search`
* See tests run and verify they all pass.
* Do the same against tablet and mobile using following command syntax - `env BROWSERSIZE=mobile ./node_modules/.bin/mocha specs/wp-support-search`.
* Read through tests and check there are no missing scenarios that might be helpful to cover.

Fixes https://github.com/Automattic/wp-calypso/issues/43135
